### PR TITLE
 Ignore low-confidence CharlockHolmes guesses when parsing link cards

### LINKED ISSUE #9466: PreviewCard does not respect charset
### Expected behaviour

Cards should parse the `Content-Type` HTTP header and `<meta http-equiv="Content-Type">` element and use the specified charset to parse the `<title>`.

### Actual behaviour

The wrong charset is used and the link preview is garbage.

### Steps to reproduce the problem

Link to https://playground.jomo.tv/windows-1251.html

Content of that site, encoded in windows-1251 (pipe it to `iconv -f utf8 -t windows-1251`):

```html
<!DOCTYPE html>
<html>
<head>
  <meta http-equiv="Content-Type" content="text/html; charset=windows-1251" />
  <title>сэмпл текст</title>
</head>
<body>
  <p>сэмпл текст</p>
</body>
</html>
```

### Specifications

Tested on a Mastodon v2.6.5 instance

### PR DESCRIPTION
**Pull request by ClearlyClaire**
Fixes #9466 by ignoring low-confidence `CharlockHolmes` charset guesses.

High-confidence guesses will still override server or document-specified charset, and low confidence guesses won't be used at all even if there is no server or document-specified charset, so this might not be the best way to proceed.

Note also that the “hint” given to `CharlockHolmes` as very little importance, and feeding it the correct charset may not prevent it from outputing an incorrect, low-confidence guess.

### CODE REVIEW

**Comment by ThibG** [comment]
Weird, the tests pass in my dev environment :thinking:

File: app/services/fetch_link_card_service.rb [hunk 1]
```diff
@@ -137,7 +137,8 @@ def attempt_opengraph
     detector.strip_tags = true
 
     guess      = detector.detect(@html, @html_charset)
-    page       = Nokogiri::HTML(@html, nil, guess&.fetch(:encoding, nil))
+    encoding   = guess&.fetch(:confidence, 0) > 60 ? guess&.fetch(:encoding, nil) : nil
```
Comment (Gargron) [inline app/services/fetch_link_card_service.rb]: I think `nil > 60` will throw no method error

**Review by Gargron — CHANGES_REQUESTED** [review]

File: app/services/fetch_link_card_service.rb (hunk 1 above)
Comment (ThibG) [reply app/services/fetch_link_card_service.rb]: Indeed! Addressed.

**Review by Gargron — APPROVED** [review]

**Comment by jomo** [comment]
Thank you, @ThibG!

**Pull request merged**